src/features/session_replay/instrument/index.js

```diff
@@ -29,6 +31,14 @@ export class Instrument extends InstrumentBase {
     }
   }
 
+  canPreloadRecorder (session) {
+    /** Assume if replay is in progress... that is enough info to know the recorder can be imported */
+    if (session?.sessionReplayMode === MODE.FULL || session?.sessionReplayMode === MODE.ERROR) return true
```

Comment:
I don't think this assumption is correct. The new page may have things disabled or APIs not avail; should run checks regardless

Reply:
what checks are you referring to here?